retroshare-gui/src/gui/statistics/TurtleRouterDialog.cpp

```diff
@@ -162,8 +162,8 @@ void TurtleRouterDialog::updateTunnelRequests(	const std::vector<std::vector<std
 
 	for(uint i=0;i<search_reqs_info.size();++i)
 	{
-		QString str = tr("Request id: %1\t from [%2]\t %3 secs ago").arg(search_reqs_info[i].request_id,0,16).arg(getPeerName(search_reqs_info[i].source_peer_id)).arg(search_reqs_info[i].age);
-
+		QString str = tr("Request id: %1\t from [%2]\t %3 secs ago\t\t %4").arg(search_reqs_info[i].request_id,0,16).arg(getPeerName(search_reqs_info[i].source_peer_id)).arg(search_reqs_info[i].age).arg(QString::fromUtf8(search_reqs_info[i].keywords.c_str()));
```

Comment:
Please pass the string length too to QString::fromUtf8(...) otherwise it will rely on the string being properly terminated with \0 to determine where the array of bytes ends, thus potentially introducing an out of bound read if the string has no ending \0 (it may happen by chance or by malicious intent)

Reply:
done, i think